docs/vpcanalyzer_lint.md

```diff
@@ -0,0 +1,93 @@
+## vpcanalyzer lint
+
+provides a detailed report of potential issues in the given VPC configuration
+
+### Synopsis
+
+The following linters are supported:
+
+| Name                            | Description                                                                              |
+|---------------------------------|------------------------------------------------------------------------------------------|
+| **nacl-split-subnet**           | Network ACLs implying different connectivity for endpoints inside a subnet               | 
+| **sg-split-subnet**             | SGs implying different connectivity for endpoints inside a subnet                        |  
+| **subnet-cidr-overlap**         | Overlapping subnet address spaces                                                        |  
+| **nacl-unattached**             | Network ACL not applied to any resources                                                 |  
+| **sg-unattached**               | SG not applied to any resources                                                          |  
+| **sg-rule-cidr-out-of-range**   | rules of security groups that reference CIDRs not in the relevant VPC address range      |
+| **nacl-rule-cidr-out-of-range** | rules of network Network ACLs that reference CIDRs not in the relevant VPC address range |
+| **tcp-response-blocked**        | Blocked TCP response                                                                     |
+| **nacl-rule-shadowed**          | rules of network Network ACLs that are shadowed by higher priority rules                 |
```

Comment:
```suggestion
| **nacl-rule-shadowed**          | Network ACL rules shadowed by higher priority rules                 |
```

Reply:
`Network ACL` or `Network-ACL` ?
We have it in several places and should be consistent
@kyorav @zivnevo

Reply:
`Network ACL`

Reply:
@zivnevo agreed? Its 2 against one :-)

Reply:
Done  (with "Network ACL")

Reply:
**TL;DR**: I don't really mind.

**Details**:
This is not the usual hyphen, it's a compound modifier. See [here](https://www.grammarbook.com/blog/definitions/punctuating-compounds-that-precede/), [here](https://www.grammarly.com/blog/hyphen-with-compound-modifiers/) and [here](https://fovndry.com/the-writers-block-a-hyphen-how-to/). So you should write `Security groups are great, but security-group rules are even better`.

Note that most places say that if there cannot be a confusion, the hyphen can be omitted (e.g., as in `chocolate chip cookies`). However, we are formalists who hate ambiguities. If there is a 1:1000 chance for someone to think that there exist a term such as "group rules", that can be either "security group-rules" or "access group-rules", we better make things straight.

Reply:
I prefer keeping it `security group` and `network ACL` - since these relies on constant in the code. ROI of keeping two constant for each is  very low.
I do not think there is even a 1:1000 chance of ambiguity here, assuming our tool will be used by persons who have some idea regarding VPC and its related constructs.